libs/server-sdk/src/data_interfaces/bootstrapper/ibootstrapper.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <launchdarkly/data_model/sdk_data_set.hpp>
+
+#include <tl/expected.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace launchdarkly::server_side::data_interfaces {
+
+/**
+ * Defines a component that can fetch a complete dataset for use in a
+ * Data System. Bootstrapping takes place when the SDK starts, and is
+ * responsible for provisioning the initial data that an SDK uses before it can
+ * begin the on-going synchronization process.
+ */
+class IBootstrapper {
+   public:
+    class Error {
```

Comment:
This is a prototype of what the IBoostrapper interface might look like and isn't final. In practice I think we'd need `FetchAll` to be asynchronous (take a completion handler.)